lib/email/utils/helpers.js

```diff
@@ -28,7 +30,7 @@ function logFlowEventFromMessage(log, message, type) {
   const elapsedTime = currentTime - flowBeginTime
 
   if (flowId && flowBeginTime && (elapsedTime > 0) && type && templateName) {
-    const eventName = `email.${templateName}.${type}`
+    const eventName = `email.${templateName}.${type}.${emailDomain}`
```

Comment:
> Or maybe we use a different delimeter than . because domains include i

Oh yes, very good point.  I find something about mixing dot-delimited strings like this quite unsettling...

> Possible sample query to get bounced, delivered and total for all templates sent to aol.

Rather than doing a separate query for each domain, I think we're more likely to want to do a `GROUP BY domain` in order to get the bounce rate for each, similar to how we do it for e.g. https://sql.telemetry.mozilla.org/queries/3426/source#6684.  Could be a lot of string parsing if we don't want to hard-code a list of domains in the query itself.

Stepping back a little...what if we didn't use flow events for this?  What if we added a separate data dump just for "email events", copying the script at [1] to dump them to S3, and updating our redshift logic to import them into a new `email_events` table?  It would be more infrastructure work but would it let us do a better/easier job of creating the graphs we want?  After all, some types of email sends are not related to a flow, and so would not show up in the flow-event-based graph above.

(That's just a hypothetical; @philbooth can probably tell me whether it's totally bananas given the amount of work involved in the existing infra)

[1] https://github.com/mozilla-services/puppet-config/blob/master/shared/modules/fxa/files/hekad/lua_outputs/fxa_flow_csv.lua